### DISCUSSION TITLE
Convert dense adj matrix to sparse matrix

### DISCUSSION THREAD

**Discussion by shrimonmuke0202** [pyg-team/pytorch_geometric#2984] [Q&A]
How can I convert dense adj matrix with shape [batch, N, N] where N is the number of nodes in the graph to its corresponding edge_index and edge_attribute?

**Comment by wsad1** [comment]
Check out [dense_to_sparse](https://pytorch-geometric.readthedocs.io/en/latest/modules/utils.html#torch_geometric.utils.dense_to_sparse).

**Reply by shrimonmuke0202** [reply]
I apply this, but It gives an error which is Assertion dim == 2

**Reply by wsad1** [reply]
Oh ok. I think the module was update to work with `dim == 2 ` or `dim ==3` from v1.7.1. Which version are you using, could you try upgrading.

**Reply by shrimonmuke0202** [reply]
Pytorch geometric is 1.7.2

**Reply by wsad1** [reply]
That's weird, cause dense_to_sparse doesn't have such an assert as you can see from the code below.
```
def dense_to_sparse(adj):
    r"""Converts a dense adjacency matrix to a sparse adjacency matrix defined
    by edge indices and edge attributes.

    Args:
        adj (Tensor): The dense adjacency matrix.
     :rtype: (:class:`LongTensor`, :class:`Tensor`)
    """
    assert adj.dim() >= 2 and adj.dim() <= 3
    assert adj.size(-1) == adj.size(-2)

    index = adj.nonzero(as_tuple=True)
    edge_attr = adj[index]

    if len(index) == 3:
        batch = index[0] * adj.size(-1)
        index = (batch + index[1], batch + index[2])

    return torch.stack(index, dim=0), edge_attr
```
